tests/linearizability/model/history.go

```diff
@@ -382,19 +382,12 @@ type History struct {
 	failed []porcupine.Operation
 }
 
-func (h History) Merge(h2 History) History {
-	result := History{
-		successful: make([]porcupine.Operation, 0, len(h.successful)+len(h2.successful)),
-		failed:     make([]porcupine.Operation, 0, len(h.failed)+len(h2.failed)),
-	}
-	result.successful = append(result.successful, h.successful...)
-	result.successful = append(result.successful, h2.successful...)
-	result.failed = append(result.failed, h.failed...)
-	result.failed = append(result.failed, h2.failed...)
-	return result
+func (h *History) Merge(h2 History) {
```

Comment:
Don't understand why making this method mutable helps anything.

Reply:
Wanted to pass the pointer of History to another function instead of copying it around. 

Is the concern of making this pointer receiver that multiple Merges mutates the history? 

I can add a lock to prevent in this case then.